server.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"gopkg.in/juju/charm.v6-unstable"
+
+	"github.com/juju/bundlechanges"
+	"github.com/julienschmidt/httprouter"
+)
+
+func getChangesFromStore(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+	bundleURL := params.ByName("bundleURL")[1:]
+	fmt.Fprintf(w, "Not implemented; bundle requested: %s", bundleURL)
+}
+
+func getChangesFromYAML(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+	bundleYAML := r.FormValue("bundleYAML")
+	if bundleYAML == "" {
+		http.Error(w, "Bundle is empty", 400)
+		return
+	}
+	bundle, err := charm.ReadBundleData(strings.NewReader(bundleYAML))
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error reading bundle data: %v", err), 422)
+		return
+	}
+	err = bundle.Verify(nil, nil)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error verifying bundle data: %v", err), 422)
+		return
+	}
+	changes := bundlechanges.FromData(bundle)
+	changesJSON, err := json.Marshal(changes)
```

Comment:
Just marshaling the returned changes isnot sufficient unfortunately. See what's done at https://github.com/juju/juju/blob/master/apiserver/client/bundles.go#L46
Also, let's try to avoid using internal package structs when the data is serialized and sent over the wire. In the linked example, we need here a type similar to params.BundleChangesChange in core.